src/vt/registry/auto/auto_registry_common.h

```diff
@@ -204,32 +144,21 @@ struct BaseScatterDispatcher {
 
 template <typename MsgT, typename HandlerT, typename ObjT>
 struct ScatterDispatcher final : BaseScatterDispatcher {
-  explicit ScatterDispatcher(HandlerT in_fn_ptr) : fn_ptr_(in_fn_ptr) {}
-
-private:
-  template <typename T, typename = void>
-  struct DispatchImpl;
-
-  template <typename T>
-  using isActiveTypedFnType = std::enable_if_t<
-    std::is_same<
-      T,
-      ActiveTypedFnType<MsgT>*
-    >::value
-  >;
-
-  template <typename T>
-  struct DispatchImpl<T, isActiveTypedFnType<T>> {
-    static void run(MsgT* msg, void*, HandlerT han) { han(msg); }
-  };
+  explicit ScatterDispatcher(HandlerT in_fn_ptr) : fp(in_fn_ptr) {}
 
 public:
-  void dispatch(void* msg, void* object) const override {
-    DispatchImpl<HandlerT>::run(static_cast<MsgT*>(msg), object, fn_ptr_);
+  void dispatch(void* msg, void*) const override {
+    using T = HandlerT;
+
+    if constexpr (std::is_same_v<T, ActiveTypedFnType<MsgT>*>) {
+      fp(static_cast<MsgT*>(msg));
+    } else {
+      vtAssert(false, "Invalid function type for scatter handler");
```

Comment:
```suggestion
      vtAbort("Invalid function type for scatter handler");
```
So it would still be there in a debug build that selected the wrong path.